src/main/java/com/piranframework/darbaan/model/Server.java

```diff
@@ -138,7 +138,7 @@ public boolean notResponding() {
    * @return server does not pass the retry count
    */
   public boolean retryPing() {
-    return --remainInterval >= 0;
+    return remainInterval - 1 >= 0;
```

Comment:
It's kind of problematic because it never updates the value of remainInterval, as you mention that remainInterval is volatile and decrementing it is not a safe action. here are two strategies we could have:
1. let deal with concurrent decrements: let it decrementing unsafe and in worse case, it issues the request in more than Retry num.
2. use AtomicInteger to guaranty that it would retry exact number
I personally prefer the first approach cause it does not block threads in a concurrent environment and retrying request is not a very precise thing to provide guaranty for.

Reply:
Got your point, in this use case there is no need to worry much about that.

Technically speaking, however, there is an issue Intellij pointed that out that the operation is not atomic.